File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02784/p02784_num9_parsed.cpp
#include <iostream>
#include <unordered_set>
 
int main() {
    int H, N;
    std::cin >> H >> N;
 
    std::unordered_set<int> moves;
 
    for (int i = 0; i < N; i++) {
        int A;
        std::cin >> A;
        moves.insert(A);
    }
 
    int remaining_health = H;
    bool can_win = false;
 
    for (int move : moves) {
        remaining_health -= move;
        if (remaining_health <= 0) {
            can_win = true;
            break;
        }
    }
 
    if (can_win) {
        std::cout << "Yes" << std::endl;
    } else {
        std::cout << "No" << std::endl;
    }
 
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02784/p02784_num9_parsed.py
import sys

def can_win(h, n, moves):
    dp = [False] * (h + 1)
    dp[0] = True

    for i in range(1, h + 1):
        for move in moves:
            if i - move >= 0 and dp[i - move]:
                dp[i] = True
                break

    return "Yes" if dp[h] else "No"

if __name__ == "__main__":
    h, n = map(int, sys.stdin.readline().split())
    moves = list(map(int, sys.stdin.readline().split()))
    print(can_win(h, n, moves))